planning/behavior_path_planner/include/behavior_path_planner/utils/path_safety_checker/objects_filtering.hpp

```diff
@@ -233,6 +249,42 @@ TargetObjectsOnLane createTargetObjectsOnLane(
 bool isTargetObjectType(
   const PredictedObject & object, const ObjectTypesToCheck & target_object_types);
 
+/**
+ * @brief Filters objects in the 'selected' container based on the provided filter function.
+ *
+ * This function partitions the 'selected' container based on the 'filter' function
+ * and moves objects that satisfy the filter condition to the 'removed' container.
+ *
+ * @tparam Func The type of the filter function.
```

Comment:
typo?

Reply:
tparam is used for template arguments. so not a typo.